src/main.cc

```diff
@@ -830,19 +845,26 @@ serverConnectionsOpen(void)
 #if SQUID_SNMP
         snmpOpenPorts();
 #endif
+    }
 
-        icmpEngine.Open();
-        netdbInit();
-        asnInit();
-        ACL::Initialize();
-        peerSelectInit();
+    neighbors_init();
+}
 
-        carpInit();
+static void
```

Comment:
If this is true:

```suggestion
/// start providing services that are only provided by workers
/// and that are unaffected by Store::Controller::WaitingForIndex()
static void
```

Reply:
Done. However, there can be many other code lines above (and below) which are "unaffected by Store::Controller::WaitingForIndex()".

Reply:
We should try to find a better definition of this new function scope/purpose then. The "and below" cases can be handled/excluded by saying that services started here should be started _before_ services that depend on WaitingForIndex(). I did not check what condition/reason can be used to justify the exclusion of the "lines above" cases.